src/Redux/Reducers-Redux/authSlice.js

```diff
@@ -1,91 +1,94 @@
-    const encodedToken=localStorage.getItem("token");
-        if(encodedToken){
-        try{
-const response=await axios.get("api/auth/verify",{encodedToken});    
-    return response.data}
-    catch(error){
-console.log(error.response);        
- }}
-});
-export const editUser=createAsyncThunk("auth/editUser",async(userData)=>{
-try {
-    const encodedToken=localStorage.getItem("token");
-    const {data}=await axios.post(`/api/users/edit`,{userData},{headers:{authorization:encodedToken}}
+export const editUser = createAsyncThunk("auth/editUser", async (userData) => {
+  try {
+    const encodedToken = localStorage.getItem("token");
+    const { data } = await axios.post(
+      `/api/users/edit`,
+      { userData },
+      { headers: { authorization: encodedToken } }
     );
-    
+
     return data;
-} catch (error) {
-  console.log(error.response.data.errors[0]);  
-}
+  } catch (error) {
+    console.log(error.response.data.errors[0]);
+  }
 });
-const authSlice=createSlice({
-    name:"auth",
-    initialState,
-    reducer:{},
-    extraReducers(builder){
-        builder
-        .addCase(login.pending,(state,action)=>{
-            state.loading=true;
-        })
-        .addCase(login.fulfilled,(state,action)=>{
-            state.currentUser=action.payload.foundUser;
-            state.loading=false;
-            localStorage.setItem("token",action.payload.encodedToken);
-            notifySuccess("Hey you loge'd in man")        
-        })
-        .addCase(signUp.fulfilled,(state,action)=>{
-            state.currentUser=action.payload.createdUser;
-             localStorage.setItem("token",action.payload.encodedToken)
-        })
-        .addCase(logOut.fulfilled,(state,action)=>{
-            notifyError("you logged out");
-            state.currentUser={};
-        })
-        .addCase(checkToken.fulfilled,(state,action)=>{
-            if(action.payload){
-            state.currentUser=action.payload.user;
-            }
-        })
-        .addCase(editUser.fulfilled,(state,action)=>{
-            state.currentUser=action.payload
-        })
-    },
+const authSlice = createSlice({
+  name: "auth",
+  initialState,
+  reducers: {
+    logOut: () => localStorage.clear(),
+  },
+  extraReducers(builder) {
+    builder
+      .addCase(login.pending, (state, action) => {
+        state.loading = true;
+      })
+      .addCase(login.fulfilled, (state, action) => {
+        state.currentUser = action.payload.foundUser;
+        state.loading = false;
+        localStorage.setItem("token", action.payload.encodedToken);
+        notifySuccess("Hey you loge'd in man");
+      })
+      .addCase(signUp.fulfilled, (state, action) => {
+        state.currentUser = action.payload.createdUser;
+        localStorage.setItem("token", action.payload.encodedToken);
+      })
+      .addCase(checkToken.fulfilled, (state, action) => {
+        if (action.payload) {
+          state.currentUser = action.payload.user;
+        }
```

Comment:
appreciation: great that you are using redux-thunk for managing user-state

Reply:
Thanks